File: CCclient/assets/Script/startButton.js
cc.Class({
    extends: cc.Component,

    properties: {
        inputPlayerName: {
            default: null,
            type: cc.Node
        },
        inputRoomId: {
            default: null,
            type: cc.Node
        }
    },

    onLoad() {
        this.gm = cc.find('gameManager').getComponent('gameManager');
    },

    onStartClicked () {
        var name = this.inputPlayerName.getComponent(cc.EditBox).string;
        var roomId = this.inputRoomId.getComponent(cc.EditBox).string;
        console.log(`name = ${name} roomId = ${roomId}`);
        if (name.length > 0) {
            this.gm.emit(reqs.LOGIN, {
                name: name,
                roomId: roomId
            });
        } else {
            // toast 
            // please enter name
        }
    }

});

File: ClientUnity/Assets/Scripts/Behaviours/StartButton.cs
using UnityEngine;
using System.Collections;
using UnityEngine.UI;
using GooglePlayGames;

public class StartButton : MonoBehaviour
{

    public Text InputPlayerName;
    public Text InputRoomId;

    public void OnClickStart()
    {
        if (InputPlayerName.text.Length > 0)
        {
            JSONObject data = JSONObject.Create();
            data.AddField("name", InputPlayerName.text);
            data.AddField("roomId", InputRoomId.text);
            GameManager.Instance.Emit(IOTypes.E_LOGIN, data);
        }
        else
        {
            // toast 
            // please enter name
        }
    }

#if UNITY_ANDROID
    public void OnLoginGoogle()
    {
        LoginScene.DispatchInitiateSpinner();
        Debug.Log("[GP]Start to Auth user with button");
        PlayGamesPlatform.Instance.localUser.Authenticate((bool success) =>
        {
            Debug.Log("[GP]button action, User login " + success);
            LoginScene.DispatchLoginGooglePlay(success);
        });
    }
#endif
}
